// @aws-sdk/client-cost-explorer/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TooManyTagsException = exports.BackfillLimitExceededException = exports.GenerationExistsException = exports.UnresolvableUsageUnitException = exports.RequestChangedException = exports.BillingViewHealthStatusException = exports.BillExpirationException = exports.DataUnavailableException = exports.InvalidNextTokenException = exports.ResourceNotFoundException = exports.UnknownSubscriptionException = exports.ServiceQuotaExceededException = exports.UnknownMonitorException = exports.LimitExceededException = exports.AnalysisNotFoundException = void 0;
const CostExplorerServiceException_1 = require("./CostExplorerServiceException");
class AnalysisNotFoundException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "AnalysisNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AnalysisNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AnalysisNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.AnalysisNotFoundException = AnalysisNotFoundException;
class LimitExceededException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "LimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.LimitExceededException = LimitExceededException;
class UnknownMonitorException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "UnknownMonitorException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "UnknownMonitorException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnknownMonitorException.prototype);
        this.Message = opts.Message;
    }
}
exports.UnknownMonitorException = UnknownMonitorException;
class ServiceQuotaExceededException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
class UnknownSubscriptionException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "UnknownSubscriptionException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "UnknownSubscriptionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnknownSubscriptionException.prototype);
        this.Message = opts.Message;
    }
}
exports.UnknownSubscriptionException = UnknownSubscriptionException;
class ResourceNotFoundException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    Message;
    ResourceName;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
        this.ResourceName = opts.ResourceName;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class InvalidNextTokenException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "InvalidNextTokenException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidNextTokenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidNextTokenException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidNextTokenException = InvalidNextTokenException;
class DataUnavailableException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "DataUnavailableException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DataUnavailableException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DataUnavailableException.prototype);
        this.Message = opts.Message;
    }
}
exports.DataUnavailableException = DataUnavailableException;
class BillExpirationException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "BillExpirationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "BillExpirationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BillExpirationException.prototype);
        this.Message = opts.Message;
    }
}
exports.BillExpirationException = BillExpirationException;
class BillingViewHealthStatusException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "BillingViewHealthStatusException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "BillingViewHealthStatusException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BillingViewHealthStatusException.prototype);
        this.Message = opts.Message;
    }
}
exports.BillingViewHealthStatusException = BillingViewHealthStatusException;
class RequestChangedException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "RequestChangedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "RequestChangedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RequestChangedException.prototype);
        this.Message = opts.Message;
    }
}
exports.RequestChangedException = RequestChangedException;
class UnresolvableUsageUnitException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "UnresolvableUsageUnitException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "UnresolvableUsageUnitException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnresolvableUsageUnitException.prototype);
        this.Message = opts.Message;
    }
}
exports.UnresolvableUsageUnitException = UnresolvableUsageUnitException;
class GenerationExistsException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "GenerationExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "GenerationExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, GenerationExistsException.prototype);
        this.Message = opts.Message;
    }
}
exports.GenerationExistsException = GenerationExistsException;
class BackfillLimitExceededException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "BackfillLimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "BackfillLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BackfillLimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.BackfillLimitExceededException = BackfillLimitExceededException;
class TooManyTagsException extends CostExplorerServiceException_1.CostExplorerServiceException {
    name = "TooManyTagsException";
    $fault = "client";
    Message;
    ResourceName;
    constructor(opts) {
        super({
            name: "TooManyTagsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTagsException.prototype);
        this.Message = opts.Message;
        this.ResourceName = opts.ResourceName;
    }
}
exports.TooManyTagsException = TooManyTagsException;
